From ddfea6b54cfa8f8f6c970d970318568c8a8a4c78 Mon Sep 17 00:00:00 2001
From: Chapman Flack <g2@anastigmatix.net>
Date: Wed, 2 Apr 2014 21:57:00 -0400
Subject: [PATCH] Break at switch_root only for bare rd.break

Previously, any rd.break=breakpoint would cause a break at the
given breakpoint and also at switch_root.
---
 modules.d/99base/init.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules.d/99base/init.sh b/modules.d/99base/init.sh
index 2364f4fccd..a1232eb5a4 100755
--- a/modules.d/99base/init.sh
+++ b/modules.d/99base/init.sh
@@ -353,7 +353,9 @@ wait_for_loginit
 # remove helper symlink
 [ -h /dev/root ] && rm -f -- /dev/root
 
-getarg rd.break -d rdbreak && emergency_shell -n switch_root "Break before switch_root"
+bv=$(getarg rd.break -d rdbreak) && [ -z "$bv" ] &&
+    emergency_shell -n switch_root "Break before switch_root"
+unset bv
 info "Switching root"
 
 
-- 
1.9.3

